### ISSUE TITLE
Overrides and Synonyms with aliases

### ISSUE THREAD

**Issue by jeanmatthieud** [typesense/typesense#621]
## Description
I'm using Typesense in production, for an e-commerce website.

I have to upgrade a collection schema. I want to use an alias, [as recommended](https://typesense.org/docs/0.23.0/api/collections.html#using-an-alias).
If this works fine for documents (I can re-index all documents), it's more difficult for overrides and synonyms.

It could be great to add, at least, a bulk synonyms / overrides import endpoint.
Better, a "migration" endpoint, to copy synonyms / overrides from a collection to another collection.

**Comment by kishorenc** [comment]
I agree that this is a bit of a pain at the moment. In the longer, I want to move synonyms and overrides top-level resources which can be "linked" to any collection.

Btw, In-place schema updates are supported in v0.23 onwards: https://typesense.org/docs/0.23.0/api/collections.html#update-or-alter-a-collection

**Comment by nickjackson** [comment]
@kishorenc this sounds good, we need this too please.

**Comment by semanticdreamer** [comment]
+1 essential

**Comment by datojije** [comment]
+1

**Comment by kishorenc** [comment]
A work around for now is to use the clone collection API which will clone a collection without document data but will copy over synonyms and overrides: https://typesense.org/docs/0.25.2/api/collections.html#cloning-a-collection-schema

**Comment by marcopesani** [comment]
+1

**Comment by sdixith** [comment]
+1

**Comment by jasonbosco** [comment]
Typesense `30.0.rca34` now has two top-level resources called "Synonym Sets" and "Curation Sets", that can be independently linked to one or more collections: 

https://typesense.org/docs/30.0/api/synonyms.html
https://typesense.org/docs/30.0/api/curation.html

So when you create a new collection, you can keep existing synonyms and curation rules as is, and just link them to the new collection.
